search.go

```diff
@@ -124,6 +129,10 @@ func (c *Client) Search(ctx context.Context, query string, t SearchType, opts ..
 	v.Set("q", query)
 	v.Set("type", t.encode())
 
+	if t == SearchTypeShow && v.Get("market") == "" {
```

Comment:
I'm not sure this is specific to searching Shows - but I'm also not sure that returning an Error here is the right thing to do. Spotify could adjust their API behaviour in future. Perhaps we just need to better document that when using Client ID/Secret you need to specify the market as there's no default market.

Reply:
agreed, better documentation is probably the way to go as we don't control the scope of these params or if/when this issue really will get fixed. we can't decode the bearer token to get user info for us to implement the fallback method can we?

Reply:
> we can't decode the bearer token to get user info for us to implement the fallback method can we?

I think we'd probably want to avoid that - even if the bearer token was decodable it'd probably be a bad idea for us to couple with that.